src/k8s/cmd/util/node_status.go

```diff
@@ -0,0 +1,37 @@
+package cmdutil
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	"github.com/canonical/k8s/pkg/client/k8sd"
+
+	"github.com/canonical/lxd/shared/api"
+)
+
+// GetNodeStatus retrieves the NodeStatus from k8sd client. If the daemon is not initialized, it exits with an error
```

Comment:
nit: it still receives the node status from the server, not the client :)

Reply:
oops sorry for that one! :smile: